syntax = "proto3";
import "protobuf/digital_id.proto";

enum Bool{
    False = 0;
    True = 1;
}

message ShareIDTransaction{
    bytes payload = 1;
    string action = 2;
}

message ShareIdRequest {

    // Bool ID_hash_requested = 1; -- changes on May 25, 2020
    Bool ID_data_requested = 1;
    bytes contract_detail = 2;
    string contract_signature = 4;
    string recv_id_from_address = 3;
}

message ShareIdResponse {

   // string digital_id_hash = 1;   // hash of the digital id in id_confirmation_txn should match
    string digital_signature = 1; // calculate signature on hash of ShareIdResponse
                                // to be verified by the id_creating_pub_key
    // bytes Id_info = 3; // not being used
    string send_to_address = 2;
}

//message Id_info{
//    string id_creating_pub_key = 1; // same as in id_confirmation_txn
//    string id_confirmation_txn = 2; // put in header dependency
//    string code = 3;
//    string contract_id = 4;
//}
message contract{
    IdAttributeSet attribute_set = 1;
    int32 validity_in_years = 2; // relative validity in years from the create_timestamp
    string create_timestamp = 3;
}

